packages/adblocker/src/engine/engine.ts

```diff
@@ -172,7 +171,10 @@ export default class FilterEngine extends EventEmitter<EngineEventHandlers> {
       return Promise.all([listsPromises, resourcesPromise]).then(([lists, resources]) => {
         const engine = this.parse(lists.join('\n'), config);
         if (resources !== undefined) {
-          engine.updateResources(resources, '' + resources.length);
+          engine.updateResources(
+            JSON.parse(resources) as ResourcesDistribution,
+            fastHash(resources).toString(16),
+          );
```

Comment:
wonder if we need to change any of that - shouldn't it be `updateResources` that does the parsing?

Reply:
We accept `string` not `ResourcesDistribution` to hide internals and keep the shape.